agent/lib/broker_address_settings.js

```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
     }
 }
 
+function addressplan_compare(a, b) {
+    return myutils.string_compare(a.name, b.name);
+}
+
 function required_broker_resource(plan) {
-    var brokerRequired = plan.requiredResources ? plan.requiredResources.filter(function (o) { return o.name === 'broker'; })[0] : undefined;
-    return brokerRequired ? brokerRequired.credit : undefined;
+    return plan.resources ? plan.resources['broker'] : undefined;
 }
 
 function same_broker_resource(a, b) {
-    return required_broker_resource(a) === required_broker_resource(b);
+    return a.name === b.name && a.resources['broker'] === b.resources['broker'];
+}
+
+function by_addressplan(a) {
+    return a.name;
+}
+
+function description(list) {
+    const max = 5;
+    if (list.length > max) {
+        return list.slice(0, max).map(by_addressplan).join(', ') + ' and ' + (list.length - max) + ' more';
+    } else {
+        return JSON.stringify(list.map(by_addressplan));
+    }
 }
 
 function BrokerAddressSettings(config) {
-    this.watcher = kubernetes.watch('configmaps', myutils.merge({selector: 'type=address-plan'}, config));
+    this.config = config || {};
+    this.selector = "";
+    events.EventEmitter.call(this);
+}
+
+util.inherits(BrokerAddressSettings, events.EventEmitter);
+
+BrokerAddressSettings.prototype.start = function () {
+    var self = this;
+    var options = myutils.merge({selector: this.selector, namespace: this.config.ADDRESS_SPACE_NAMESPACE}, this.config);
+    this.watcher = kubernetes.watch('addressplans', options);
     this.watcher.on('updated', this.updated.bind(this));
     this.required_broker_resource = {};
-    this.last = undefined;
+    this.last = {};
+    this.get_global_max_size().then(function (size) {
+        self.global_max_size = size;
+    });
 }
 
-BrokerAddressSettings.prototype.wait_for_plans = function () {
-    var self = this;
-    return new Promise(function (resolve, reject) {
-        self.watcher.once('updated', function () {
-            log.info('plans have been retrieved');
+BrokerAddressSettings.prototype.get_changes = function (name, addressplans, unchanged) {
+    var c = myutils.changes(this.last[name], addressplans, addressplan_compare, unchanged, description);
+    this.last[name] = addresses;
+    return c;
+};
+
+BrokerAddressSettings.prototype.get_global_max_size = function () {
```

Comment:
I think a simpler way to get this would be to inject the value as an environment variable in the template..